### DISCUSSION TITLE
Hi how to get the privateKey after generate an account from Mnemonic

### DISCUSSION THREAD

**Discussion by francelwebdev** [wevm/viem#1487] [General]
Hi how to get the privateKey after generate an account from Mnemonic

**Comment by jxom** [comment]
Viem doesn't expose an API to extract private keys (for security reasons). You can rely on an external **audited** library, like `@scure/bip32` and `@scure/bip39` for this.

```tsx
import { HDKey } from '@scure/bip32'
import { mnemonicToSeedSync } from '@scure/bip39'

const seed = mnemonicToSeedSync("foo bar baz ...")
const { privateKey } = HDKey.fromMasterSeed(seed)
```

**Comment by itxtoledo** [comment]
```typescript
const wallet = mnemonicToAccount(mnemonic);
const eoaPrivateKey = wallet.getHdKey().privateKey;
const eoaPublicAddress = wallet.address;
```

**Comment by christianalberto** [comment]
HI @francelwebdev,  depending on the version of Viem you are using, you can achieve this either natively or via Viem's underlying audited crypto dependencies (`@scure/bip32` / `@scure/bip39`).

Here is a breakdown of both approaches:

---

### Option 1: Native Viem Helper (`mnemonicToAccount`)

In recent versions of Viem, you can extract the private key directly from the `HDKey` instance returned by `mnemonicToAccount`:

```typescript
import { mnemonicToAccount } from 'viem/accounts';
import { bytesToHex } from 'viem';

const account = mnemonicToAccount('your mnemonic phrase here ...');

// Obtain HDKey and derive the private key bytes
const hdKey = account.getHdKey();
const privateKeyBytes = hdKey.privateKey;

if (privateKeyBytes) {
  // Convert Uint8Array to 0x-prefixed hex string
  const privateKey = bytesToHex(privateKeyBytes);
  console.log('Private Key:', privateKey);
}
```


Option 2: Low-Level Derivation (@scure/bip32 + @scure/bip39)

If you are using an older Viem release or prefer to derive the keys independently using the standard Ethereum derivation path (m/44'/60'/0'/0/0):

```TypeScript
import { HDKey } from '@scure/bip32';
import { mnemonicToSeedSync } from '@scure/bip39';
import { bytesToHex } from 'viem';

const seed = mnemonicToSeedSync('your mnemonic phrase here ...');
const hdKey = HDKey.fromMasterSeed(seed);

// Derive standard Ethereum path
const accountKey = hdKey.derive("m/44'/60'/0'/0/0");

if (accountKey.privateKey) {
  const privateKey = bytesToHex(accountKey.privateKey);
  console.log('Private Key:', privateKey);
}
```
⚠️ Security Note: Avoid logging or persisting raw private keys in client-side code, logs, or unencrypted storage.
